platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,255 @@
+# Merkle State API
+
+## Summary
+
+Refactor the Merkle tree so that it can be fully represented by the State API objects.
+Create a module that will provide a simple and easy-to-use API for the Merkle tree.
+
+## Purpose and Context
+
+### Why?
+
+The Merkle tree is a rather complex data structure, and to make interaction easier, we need to hide this complexity behind the API.
+Another reason to have this API is to provide additional flexibility in changing the implementation details of the Merkle tree.
+As long as they are hidden, we are free to change them. This API will be critically important for the Block Node to 
+interact with the state without knowing the details of the Merkle tree.
+
+Note that this API is already implemented as a part of the Service Modularization project, but these classes will be moved
+to a separate module as a part of this proposal.
+
+### Context
+
+There are two issues that need to be addressed:
+
+* We need to have a public API for the Merkle tree that is simple and easy to use, residing in a separate module with a minimal set of dependencies.
+* The Merkle tree has to be fully expressed in terms of State API. Currently, `PlatformState` stands out and needs to be refactored.
+
+#### Singleton
+
+A singleton is a node with 2 children:
+
+- string label for State Name
+- protobuf message for child data
+
+In Java code it's represented by the following classes:
```

Comment:
Does `String getStateKey()` references the name and `T get()` the protobuf message? Otherwise I do not see how this is related the description.

Reply:
I added the javadoc for these interfaces to make it clearer.

Reply:
> label of type String for the state name

`getStateKey()`

> protobuf message for child data

`get()`


I will modify the description to hopefully add some more clarity